data-prepper-plugins/opensearch/src/test/java/org/opensearch/dataprepper/plugins/sink/opensearch/ServerlessNetworkPolicyUpdaterTest.java

```diff
@@ -0,0 +1,360 @@
+
+        // Assertions
+        assertThat(result.get(DESCRIPTION).asString(), equalTo("Created by Data Prepper"));
+
+        // Asserting the RULES section
+        final List<Document> rulesList = result.get(RULES).asList();
+        assertThat(rulesList.size(), equalTo(1));
+        final Map<String, Document> rule = rulesList.get(0).asMap();
+        assertThat(rule.get(RESOURCE_TYPE).asString(), equalTo(COLLECTION));
+        final List<Document> collectionList = rule.get(RESOURCE).asList();
+        assertThat(collectionList.size(), equalTo(1));
+        final String expectedCollectionValue = String.format("%s/%s", COLLECTION, collectionName);
+        assertThat(collectionList.get(0).asString(), equalTo(expectedCollectionValue));
+
+        // Asserting the SOURCE_VPCES section
+        final List<Document> sourceVpcesList = result.get(SOURCE_VPCES).asList();
+        assertThat(sourceVpcesList.size(), equalTo(1));
+        assertThat(sourceVpcesList.get(0).asString(), equalTo(vpceId));
+    }
+
+    @Test
+    public void testMatchesPatternExactMatch() {
+        final String pattern = "exactMatch";
+        final String value = "exactMatch";
+
+        assertTrue(ServerlessNetworkPolicyUpdater.matchesPattern(pattern, value));
+    }
+
+    @Test
+    public void testMatchesPatternOnlyWildcard() {
+        final String pattern = "*";
+        final String value = "anyValueCanGoHere";
+
+        assertTrue(ServerlessNetworkPolicyUpdater.matchesPattern(pattern, value));
+    }
+
+    @Test
+    public void testMatchesPatternPrefixPlusWildcard() {
+        final String pattern = "prefix*";
+        final String value = "prefixWithSomeOtherStuff";
+
+        assertTrue(ServerlessNetworkPolicyUpdater.matchesPattern(pattern, value));
+    }
+
+    @Test
+    public void testMatchesPatternWildcardPlusSuffix() {
+        final String pattern = "*Suffix";
+        final String value = "thisIsThePrefixSuffix";
+
+        assertTrue(ServerlessNetworkPolicyUpdater.matchesPattern(pattern, value));
+    }
+
+    @Test
+    public void testDoesNotMatchPatternExactMatch() {
+        final String pattern = "exactMatch";
+        final String value = "notExactMatch";
+
+        assertFalse(ServerlessNetworkPolicyUpdater.matchesPattern(pattern, value));
+    }
+
+    @Test
+    public void testDoesNotMatchPatternPrefixPlusWildcard() {
+        final String pattern = "prefix*";
+        final String value = "thisIsNotTheRightPrefixWithSomeOtherStuff";
+
+        assertFalse(ServerlessNetworkPolicyUpdater.matchesPattern(pattern, value));
+    }
+
+    @Test
+    public void testDoesNotMatchPatternWildcardPlusSuffix() {
+        final String pattern = "*suffix";
+        final String value = "thisIsThePrefixButNotTheRightSuffix";
+
+        assertFalse(ServerlessNetworkPolicyUpdater.matchesPattern(pattern, value));
+    }
+
+    @Test
+    public void testHasAcceptablePolicyBothConditionsTrue() {
```

Comment:
Ideally, tests verify the correct behavior. What really matters for this test is to not create a policy when the policy is acceptable. These tests are only testing a private method, not the actual behavior.

You can improve these tests by stubbing the document, calling `updateNetworkPolicy` and verifying that no updates occur.

You could use a `@Nested` class to perform a lot of the common setup that the tests will need.

Reply:
Yea, I think that is better. Make the methods private and delete any extra tests. If anything is tested there, you can move it into a public test as necessary.